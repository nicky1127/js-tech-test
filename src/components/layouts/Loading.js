import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { makeStyles } from "@material-ui/core/styles";
import {
  CircularProgress
} from "@material-ui/core";
import labels from "constants/labels";

const constants = labels.Loading;

const useStyles = makeStyles(theme => ({
  root: {
    width: "100%",
    height: "100%",
    position: "absolute",
    zIndex: "300",
    backgroundColor: "rgba(100,100,100,0.7)",
    display: "flex",
    justifyContent: "center"
  },
  progress: {
    position: "absolute",
    top: "50vh"
  }
}));

export const Loading = props => {
  const classes = useStyles();

  return (
    <div className={clsx("loading", classes.root)}>
      <CircularProgress className={classes.progress} />
    </div>
  );
};

export default Loading;
