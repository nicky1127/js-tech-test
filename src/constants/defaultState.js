const state = {
  liveEvents: [],
  event: {},
  markets: {},
  isOddsDecimal: false,
  loading: false,
  error: {},
  selections: []
};

export default state;
