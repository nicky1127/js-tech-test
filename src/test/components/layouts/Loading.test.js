import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Loading from "components/layouts/Loading";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import defaultState from "constants/defaultState";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = {};

const state = {
  ...defaultState
};
const mockStore = configureMockStore();
const store = mockStore(state);

describe("Loading", () => {
  it("Snapshot matchs", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Loading {...props} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Loading should render one [data-testid=loading]", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Loading {...props} />
      </Provider>
    );
    const container = getByTestId("loading");
    expect(container).toBeInTheDocument();
  });
});
