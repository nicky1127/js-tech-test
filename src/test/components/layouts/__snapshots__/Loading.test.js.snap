// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Snapshot matchs 1`] = `
<div
  className="loading makeStyles-root-1"
  data-testid="loading"
>
  <div
    className="MuiCircularProgress-root makeStyles-progress-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
